sedaprotocol/seda-rust
# Add ability to remotely add/remove/list peers

## Motivation

A node operator should be able to add/remove/list peers on the fly through the CLI. This implements those functions

## Explanation of Changes



Move known_peers up a level so we can pass it to the RPC and P2PServer through Arc<RwLock<_>>. Also added the commands in the cli package and implement the P2PCommand (mpsc channel)

## Testing



1. Start two nodes
2. Use `seda node peers list` to see the connected peers
3. Use `seda node peers remove <PEER_ID>` to remove a peer (PEER_ID can be found by using list)
4. Use `seda node peers list` to see that it has been removed
5. Use `seda node peers add <MULTI_ADDR>` to add the peer back again (MULTI_ADDR) can be found in the logs from the other node. (looks like "/ip4/172.19.0.2/tcp/44681")
